public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.test.ts

```diff
@@ -797,4 +802,126 @@ describe('IN_LOGFMT completions', () => {
       ]
     `);
   });
+
+  describe('calculateRange', () => {
+    let monaco: Monaco;
+    beforeEach(() => {
+      monaco = {
+        Range: {
+          lift(range: monacoTypes.Range): monacoTypes.Range {
+            return range;
+          },
+        },
+      } as Monaco;
+    });
+
+    it('getSituation fails to return autocomplete when inserting before any other labels', () => {
+      // Note this is not desirable functionality
```

Comment:
I'm not sure I understand the comment. In the sense of we're asserting that it doesn't misbehave, or that the output is not desirable.

Reply:
Yeah I'll update this comment, basically these are all situations that autocomplete should handle better, but currently doesn't behave as we'd expect. Basically if someone made a change and these tests fail, that would be good.